autoblocks/_impl/api/client.py

```diff
@@ -230,3 +235,129 @@ def get_dataset(
             revision_id=resp["revisionId"],
             items=[DatasetItem(id=item["id"], splits=item["splits"], data=item["data"]) for item in resp["items"]],
         )
+
+    def get_local_test_runs(self, test_external_id: str) -> List[AutoblocksTestRun]:
+        req = self._client.get(f"/testing/local/tests/{encode_uri_component(test_external_id)}/runs")
+        req.raise_for_status()
+        resp = req.json()
+        return [AutoblocksTestRun(id=run["id"]) for run in resp["runs"]]
+
+    def get_ci_test_runs(self, test_external_id: str) -> List[AutoblocksTestRun]:
+        req = self._client.get(f"/testing/ci/tests/{encode_uri_component(test_external_id)}/runs")
+        req.raise_for_status()
+        resp = req.json()
+        return [AutoblocksTestRun(id=run["id"]) for run in resp["runs"]]
+
+    def get_local_test_results(self, run_id: str) -> List[AutoblocksTestCaseResultId]:
+        req = self._client.get(f"/testing/local/runs/{encode_uri_component(run_id)}/results")
+        req.raise_for_status()
+        resp = req.json()
+        return [AutoblocksTestCaseResultId(id=result["id"]) for result in resp["results"]]
+
+    def get_ci_test_results(self, run_id: str) -> List[AutoblocksTestCaseResultId]:
+        req = self._client.get(f"/testing/ci/runs/{encode_uri_component(run_id)}/results")
+        req.raise_for_status()
+        resp = req.json()
+        return [AutoblocksTestCaseResultId(id=result["id"]) for result in resp["results"]]
+
+    def get_local_test_result(self, test_case_result_id: str) -> AutoblocksTestCaseResult:
+        req = self._client.get(f"/testing/local/results/{encode_uri_component(test_case_result_id)}")
+        req.raise_for_status()
+        resp = req.json()
+        result = resp["testCaseResult"]
+
+        # Convert events to Event objects
+        events = [
+            Event(
+                id=event.get("id"),
```

Comment:
we should update the usages of get to only when we expect something to be able to be set for None, that way we through an error when something unexpected happens.

For example id should always be there, but properties we can keep as get with a default. Same with the other usages of `.get`

Reply:
@adamnolte so, do you mean for example, as we always expect id, i do
 id=event["id"] instead, and for all the other non-optional fields